builder/slide_and_fold_storage.cc

```diff
@@ -369,17 +378,36 @@ class slide_and_fold : public node_mutator {
           // effectively not slide while running before the original loop min.
           (*bounds)[d].min = select(loop_var <= loop.orig_min, old_min, new_min);
         }
-      } else if (prove_true(ignore_loop_max(is_monotonic_decreasing))) {
+      } else if (prove_true(is_monotonic_decreasing)) {
         // TODO: We could also try to slide when the bounds are monotonically
         // decreasing, but this is an unusual case.
       }
     }
   }
 
   template <typename T>
-  void visit_call_or_copy(const T* op, span<const var> outputs) {
+  void visit_call_or_copy(const T* op, span<const var> inputs, span<const var> outputs) {
     set_result(op);
 
+    for (var input : inputs) {
```

Comment:
I've been thinking we can just skip calling slide_and_fold_buffer if buffer for a given crop has consumers outside of the crop, but I think this approach works too,

Reply:
Discussed offline, let's leave this approach for now. I think it's a bit messy but I also don't know of a better way that is less messy.